### DISCUSSION TITLE
PrimeNG using dock click shows context menu

### DISCUSSION THREAD

**Discussion by vnomohan** [primefaces/community#339] [PrimeNG] [answered]
Im using angular2 installed with npm primeNG

My question is how to use the dock icon double click and show the context menu as desired. As illustrated in below picture.
`https://stackoverflow.com/questions/77371134/primeng-using-dock-click-shows-context-menu`

Any help or advise will be really appreciated. Thanks

**Comment by cetincakiroglu** [comment]
Hi,

Could you please share a stackblitz example so we can identify the issue clearly?

**Reply by vnomohan** [reply]
hi Cetincakiroglu,
I dont have the working code yet. in the provided link screenshot it was dummy and edited so that will get clear picture on what i needed.
sorry for the confusion. 
As i checked in primeng dock official website i dont fine anything regarding on this.

**Reply by cetincakiroglu** [reply]
Hi,

It's not a context menu it's a tree component and it's included in the dock demo. It's displayed when the user presses on `finder` icon to represent a folder tree. So it's not a bug, it's just to show how to use with other components as an advanced example.

**Reply by vnomohan** [reply]
hi cetincakiroglu,
Yes you are correct. but my question was how to use the context menu on dock item. it is not describe in the primeng documentation.
Also i did not say that it is a bug though. the screenshot i shared was is only for illustration to get clear picture what i need.
I hope you get this time.

**Reply by cetincakiroglu** [reply, marked as answer]
Hi,

You can trigger contextmenu on double click by using triggerEvent property `triggerEvent="dblclick"`, unfortunately you cannot use contextmenu as you've described in the dock.

**Reply by vnomohan** [reply]
hi cetincakiroglu,
Got it, let me check further on this. i am also thinking if this not able to trigger directly from dock calling the context menu.... how about if in middle calling another method and pass to context menu. Is this achievable?
i have checked from our end as well. nothing is helpfull. we really wanted to use the dock as that is attractive in ui however seems limited attributes.. :(

**Comment by vnomohan** [comment]
I believe there is NO alternative solution. Thanks anyway!
